hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -493,27 +507,20 @@ private void callDisconnectionOperation(ClientEndpointImpl endpoint) {
                 // "a disconnected client is reconnected back to same node"
                 return;
             }
-            ClientDisconnectionOperation op = createClientDisconnectionOperation(clientUuid, memberUuid);
-            operationService.run(op);
+            int partitionId = nodeEngine.getPartitionService().getPartitionId(clientUuid);
 
             for (Member member : memberList) {
-                if (!member.localMember()) {
-                    op = createClientDisconnectionOperation(clientUuid, memberUuid);
-                    operationService.send(op, member.getAddress());
+                ClientDisconnectionOperation op = new ClientDisconnectionOperation(clientUuid, memberUuid);
+                Future future = operationService.createInvocationBuilder(SERVICE_NAME, op, member.getAddress()).invoke();
+                try {
+                    future.get();
```

Comment:
Why blocking is needed here? As far as I see, `callDisconnectionOperation(..)` is called from a scheduler thread.

Reply:
And `ClientDisconnectionOperation()` offloads the actual logic to a separate thread, what blocking gives here?